Tightening trait bounds

filippodebortoli opened the pull request:
Addresses #87 

Goal: minimize usage of `model::ForIRI` and `ontology::indexed::ForIndex`.

Extras:
+ [x] fixes clippy warnings for `ontology::*` ~and top-level modules~.
+ [x] adds Deref<Target = str> to ForIRI (temporarily).
+ [ ] ~ran `cargo fmt` over the codebase~ (reverted).

phillord committed:
Delete collection which was unused

phillord commented:
We really need to unpick the cargo fmt changes which have introduced a lot of diffs. It makes it very hard to see what has been done for the trait bounds.

We should run cargo fmt before the 1.0 release, I think, and then put a check into the git hooks to standardize formatting in future. But, we need to do this in a different commit.

filippodebortoli committed:
Begins tightening of trait bounds regarding `ForIRI` and `ForIndex`.

filippodebortoli committed:
Begins tightening of trait bounds regarding `ForIRI` and `ForIndex`.

filippodebortoli commented:
> We really need to unpick the cargo fmt changes which have introduced a lot of diffs. It makes it very hard to see what has been done for the trait bounds.
> 
> We should run cargo fmt before the 1.0 release, I think, and then put a check into the git hooks to standardize formatting in future. But, we need to do this in a different commit.

Done (hopefully right), I only committed files relevant to the PR this time.

filippodebortoli committed:
Completes bound tightening for `model`.
The definition of `ForIRI` is still in place and `MutableOntology` demands `A: ForIRI`.

filippodebortoli committed:
Completes bound tightening for `model`.
The definition of `ForIRI` is still in place and `MutableOntology` demands `A: ForIRI`.

filippodebortoli committed:
Tightens trait bounds in `vocab`.

filippodebortoli committed:
Tightens trait bounds in `vocab`.

filippodebortoli committed:
Adds some documentation to `model::ForIRI` and `model::Build`.

filippodebortoli committed:
Adds a unit test for builders using Strings to create IRIs.
There is something in (&*iri_string) that could potentially be investigated.

filippodebortoli committed:
Rebasing from devel and fixing mistakes.

filippodebortoli committed:
Minimizes usage of as_ref() where possible.
Waiting for new release of pretty_rdf to remove AsRef from ForIRI.

filippodebortoli committed:
Merge branch 'devel' into feature/tighten-trait-bounds

filippodebortoli committed:
Fixes broken doctest and clippy.

phillord commented:
I have updated https://github.com/phillord/horned-owl/issues/87 with my thoughts on this.
